src/components/App.tsx

```diff
@@ -3827,6 +3827,14 @@ class App extends React.Component<AppProps, AppState> {
 
   private handleAppOnDrop = async (event: React.DragEvent<HTMLDivElement>) => {
     try {
+      if (this.props.onDrop) {
+        try {
+          await this.props.onDrop(event);
+          return;
```

Comment:
looks like right now the drop action will always be prevented (since we just return ) if present which is incorrect. We should also pass the data which is dropped similar to how we do in `onPaste` so host can use it if needed.
Right now we are preventing paste action if the `onPaste` returns true but I do agree with @dwelle we should allow `false` instead to prevent to be semantically more correct.
We can make the `onPaste` change in a separate PR, let's fix the `onDrop` in this PR.